src/Microsoft.Security.Utilities.Core/SecretMasker.cs

```diff
@@ -348,23 +393,35 @@ public IEnumerable<Detection> DetectSecrets(string input)
             yield break;
         }
 
-        if (RegexPatterns.Count == 0 &&
-            EncodedSecretLiterals.Count == 0 &&
-            ExplicitlyAddedSecretLiterals.Count == 0)
-        {
-            yield break;
-        }
-
```

Comment:
This should have been in the read lock, but this condition to check if we have a degenerate secret masker with nothing added to it doesn't seem worth optimizing for.